src/System.CommandLine.Subsystems/Subsystem/CliSubsystem.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics.CodeAnalysis;
+
+namespace System.CommandLine.Subsystems;
+
+public abstract class CliSubsystem
+{
+    protected CliSubsystem(string name, IAnnotationProvider? annotationProvider, SubsystemKind subsystemKind)
+    {
+        Name = name;
+        _annotationProvider = annotationProvider;
+        SubsystemKind = subsystemKind;
+    }
+
+    /// <summary>
+    /// The name of the extension. 
+    /// </summary>
+    public string Name { get; }
+    public SubsystemKind SubsystemKind { get; }
+
+    DefaultAnnotationProvider? _defaultProvider;
+    readonly IAnnotationProvider? _annotationProvider;
```

Comment:
nit: We have typically specified access modifiers.
```suggestion
    private DefaultAnnotationProvider? _defaultProvider;
    private readonly IAnnotationProvider? _annotationProvider;
```